rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/nodesource/NodeSourceWindow.java

```diff
@@ -492,7 +492,11 @@ private void resetFormForInfrastructureSelectChange() {
             latestPoliciesList = sorted;
             policySelectItem.setValueMap(sorted);
             if (!appropriatePolicies.contains(policySelectItem.getValueAsString())) {
-                policySelectItem.setValue("");
+                if (policySelectItem.getValueAsString() != null) {
+                    setNodeSourceWindowLabelWithError(policySelectItem.getValueAsString() + " is not compatible with " +
+                                                      infrastructurePluginName + ".");
+                }
+                policySelectItem.setValue((String) null);
```

Comment:
This cast is really weird. Is it really required ?

Reply:
Yeah. Otherwise, it is ambiguous: setValue(Double) or setValue(String)?